src/store/middlewares/savedAccounts.js

```diff
@@ -27,10 +29,43 @@ const savedAccountsMiddleware = (store) => {
     }
   });
 
+  const updateSavedAccounts = (peers, tx, savedAccounts) => {
+    const { accounts } = savedAccounts;
+    tx.forEach((transaction) => {
```

Comment:
if there are 10 transactions of an account in one block, then the account is fetched 10 times. I know it's unlikely, but still.